schema/schema.go

```diff
@@ -54,18 +54,42 @@ func F(value float64) *float64 {
 // getTagValue returns a value of the schema's type for the given tag string.
 // Uses JSON parsing if the schema is not a string.
 func getTagValue(s *Schema, t reflect.Type, value string) (interface{}, error) {
+	// Special case: strings don't need quotes.
 	if s.Type == "string" {
 		return value, nil
 	}
 
+	// Special case: array of strings with comma-separated values and no quotes.
+	if s.Type == "array" && s.Items != nil && s.Items.Type == "string" && value[0] != '[' {
+		values := []string{}
+		for _, s := range strings.Split(value, ",") {
+			values = append(values, strings.TrimSpace(s))
+		}
+		return values, nil
+	}
+
 	var v interface{}
 	if err := json.Unmarshal([]byte(value), &v); err != nil {
 		return nil, err
 	}
 
+	vv := reflect.ValueOf(v)
 	tv := reflect.TypeOf(v)
 	if v != nil && tv != t {
-		if !tv.ConvertibleTo(t) {
+		if tv.Kind() == reflect.Slice {
+			// Slices can't be cast due to the different layouts. Instead, we make a
+			// new instance of the destination slice, and convert each value in
+			// the original to the new type.
+			tmp := reflect.MakeSlice(t, 0, vv.Len())
```

Comment:
I really appreciate this comment here.  It's more than slightly confusing what's going on here!